crypto/evp/evp_lib.c

```diff
@@ -298,6 +298,14 @@ int EVP_CIPHER_CTX_nid(const EVP_CIPHER_CTX *ctx)
 
 int EVP_MD_block_size(const EVP_MD *md)
 {
+    if (md == NULL) {
+        EVPerr(EVP_F_EVP_MD_BLOCK_SIZE, EVP_R_MESSAGE_DIGEST_IS_NULL);
+        return -1;
+    }
```

Comment:
We really need to make up our minds on if we should check for NULL input or not.

Reply:
+1 for checking for _public_ APIS.

Reply:
I'm ambivalent.  Ponder unchecked errors, the caller gets -1 back and tries to actually use it.  At some point, it becomes a huge number in a `size_t` and we have something failing *somewhere else* because of it, making the initlal error (calling this function with a NULL) argument harder to find.

But then again, we have banned hard asserts, and not checking NULL here *is* a hard assert, so we're sending a mixed message already.

Like I said, I'm ambivalent...

Reply:
We *do* allow soft asserts, though, so this could be rewritten using our usual `ossl_assert` construction.  Could that be a compromise to consider?